pkg/version/info.go

```diff
@@ -0,0 +1,58 @@
+package version
+
+import (
+	"bytes"
+	"runtime"
+	"strings"
+	"text/template"
+)
+
+// These are mostly set during compilation from linker flags
+var (
+	application = "tuwat"
+	version     = "dev"
+	revision    string
+	branch      string
+	buildDate   string
+)
+
+// A rich representation of the version information which can also be readily serialized into a JSON representation
+type VersionInfo struct {
+	Application string `json:"application"`
+	Version     string `json:"version"`
+	Revision    string `json:"revision,omitempty"`
+	Branch      string `json:"branch,omitempty"`
+	BuildDate   string `json:"buildDate,omitempty"`
+	GoVersion   string `json:"goVersion"`
+	GoPlatform  string `json:"goPlatform"`
+}
+
+var versionInfoTmpl = `
+{{.Application}}, version {{.Version}} (branch: {{.Branch}}, revision: {{.Revision}})
+  build date:       {{.BuildDate}}
+  go version:       {{.GoVersion}}
+  platform:         {{.GoPlatform}}
+`
```

Comment:
Unsure who the recipient of this information is.  For running dev versions this seems good.  For requesting the version output in bug report this seems good.  The branch and revision aren't really needed for tagged versions.  It just seems a little overloaded with information.
The build date should be the release date I guess if we retain that field.

Reply:
I realized from other tools that i use, that less may be more and it is now of the following form: `tuwat v0.8.1-dev-47df78f (release date: 2022-10-31T15:58:59Z)` (see 6ca140f)